homeassistant/config_entries.py

```diff
@@ -3334,15 +3334,24 @@ async def async_finish_flow(
         if unique_id is not None and not isinstance(unique_id, str):
             raise HomeAssistantError("unique_id must be a string")
 
-        self.hass.config_entries.async_add_subentry(
-            entry,
-            ConfigSubentry(
-                data=MappingProxyType(result["data"]),
-                subentry_type=subentry_type,
-                title=result["title"],
-                unique_id=unique_id,
-            ),
-        )
+        try:
+            self.hass.config_entries.async_add_subentry(
+                entry,
+                ConfigSubentry(
+                    data=MappingProxyType(result["data"]),
+                    subentry_type=subentry_type,
+                    title=result["title"],
+                    unique_id=unique_id,
+                ),
+            )
+        except data_entry_flow.AbortFlow as err:
+            return SubentryFlowResult(
+                type=data_entry_flow.FlowResultType.ABORT,
+                flow_id=flow.flow_id,
+                handler=flow.handler,
+                reason=err.reason,
+                description_placeholders=err.description_placeholders,
+            )
```

Comment:
We may want to have this error handling here in the base class instead, since other classes may also want to abort late:
https://github.com/home-assistant/core/blob/8ca3a9297fda02d0f3727ba899454aebaa90b24e/homeassistant/data_entry_flow.py#L550

In particular, we want to remove the pattern where integrations on purpose create duplicated config entries instead of implementing option flow or reconfigure flow: https://developers.home-assistant.io/blog/2025/03/01/config-flow-unique-id/

Reply:
Sounds good to do it in the base class.